src/current/v24.3/logging-best-practices.md

```diff
@@ -0,0 +1,91 @@
+## Consider workload performance impact
+
+Enabling logging of events in the following [log channels]({% link {{ page.version.version }}/logging.md %}#logging-channels) may have a performance impact:
+
+- [`SQL_EXEC`]({% link {{ page.version.version }}/logging-use-cases.md %}#sql_exec): Logging cluster-wide executions by enabling the [`sql.trace.log_statement_execute` cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}#setting-sql-trace-log-statement-execute) will incur considerable overhead and may have a negative performance impact.
+- [`SQL_PERF`]({% link {{ page.version.version }}/logging-use-cases.md %}#sql_perf): Setting [`sql.log.slow_query.latency_threshold`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-server-auth-log-sql-sessions-enabled) to a non-zero time enables tracing on all queries, which impacts performance. After debugging, set the value back to `0s` to disable the log.
+- [`SESSIONS`]({% link {{ page.version.version }}/logging-use-cases.md %}#sessions): Logging client connection and session authentication events are enabled by the [`server.auth_log.sql_connections.enabled` cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}#setting-server-auth-log-sql-connections-enabled) and the [`server.auth_log.sql_sessions.enabled`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-server-auth-log-sql-sessions-enabled) cluster setting respectively. These logs perform one disk I/O per event. Enabling each setting will impact performance.
+- [`SENSITIVE_ACCESS`]({% link {{ page.version.version }}/logging-use-cases.md %}#sensitive_access): Enabling [Table-based SQL Audit Logging]({% link {{ page.version.version }}/sql-audit-logging.md %}) or [Role-based SQL Audit Logging]({% link {{ page.version.version }}/role-based-audit-logging.md %}) can negatively impact performance. Log only what you require to limit impact to your workload. Use this channel for [security purposes](#use-logs-for-security-and-compliance) only.
+
+## Use structured logs
+Use structured logs when you need to capture detailed, machine-readable information that can be easily parsed and analyzed. They are particularly useful for:
+
+- [Notable Events]({% link {{ page.version.version }}/eventlog.md %}): Capturing significant events in a structured format allows for easier querying and analysis. For example, job state changes or specific events like backups.
+- [Audit and Compliance]({% link {{ page.version.version }}/logging-use-cases.md %}#security-and-audit-monitoring): Structured logs are ideal for audit trails and compliance monitoring because they provide a consistent format that can be easily searched and filtered.
+- [Performance Monitoring]({% link {{ page.version.version }}/logging-use-cases.md %}#performance-tuning): When tracking performance metrics, structured logs can help in identifying patterns and anomalies due to their detailed and consistent format.
+
+## Use unstructured logs
```

Comment:
The way this is worded makes me think that customers have a choice whether they can use structured or unstructured logs, when in reality structured vs unstructured logging is a decision we (CRL) make. Would it make more sense to phrase these sections in a way that describes that we have these two different types of logs that are used in the way you described?

Reply:
updated